packages/backend-defaults/src/entrypoints/rootConfig/createConfigSecretEnumerator.ts

```diff
@@ -19,18 +19,68 @@ import type { Config } from '@backstage/config';
 import { ConfigSchema, loadConfigSchema } from '@backstage/config-loader';
 import { getPackages } from '@manypkg/get-packages';
 
+import path from 'path';
+import fs from 'fs';
+
+// Looks for a package.json with a workspace config to identify the root of the monorepo
+export function findRootPath(searchDir: string): string | undefined {
+  let currentPath = searchDir;
+
+  // Some confidence check to avoid infinite loop
+  for (let i = 0; i < 1000; i++) {
+    const packagePath = path.resolve(currentPath, 'package.json');
+    const exists = fs.existsSync(packagePath);
+    if (exists) {
+      return currentPath;
+    }
+
+    const newPath = path.dirname(currentPath);
+    if (newPath === currentPath) {
+      return undefined;
+    }
+    currentPath = newPath;
+  }
+
+  throw new Error(
+    `Iteration limit reached when searching for root package.json at ${searchDir}`,
+  );
+}
+
 /** @public */
 export async function createConfigSecretEnumerator(options: {
   logger: LoggerService;
   dir?: string;
   schema?: ConfigSchema;
 }): Promise<(config: Config) => Iterable<string>> {
-  const { logger, dir = process.cwd() } = options;
-  const { packages } = await getPackages(dir);
+  const { logger, dir } = options;
+  let closestPackagePath = process.cwd();
+  if (dir) {
+    closestPackagePath = dir;
+  } else {
+    const rootPath = findRootPath(
+      path.resolve(process.argv[1]) ?? process.cwd(),
+    );
+    if (rootPath) {
+      closestPackagePath = rootPath;
```

Comment:
Should we perhaps fall back to the old behavior for this case instead? just to be sure